### DISCUSSION TITLE
how use on raspberry

### DISCUSSION THREAD

**Discussion by mevrard31** [bbernhard/signal-cli-rest-api#396] [Q&A]
Hello,
I'm start the docker with this command on my raspberry
docker run --platform linux/arm64 -d --name signal-api --restart=always -p 8080:8080       -v $HOME/.local/share/signal-cli:/home/.local/share/signal-cli       -e 'MODE=normal' bbernhard/signal-cli-rest-api
i have a picture with this link: http://localhost:8080/v1/qrcodelink?device_name=signal-api

but when a send a message with this:
curl -X POST -H "Content-Type: application/json" 'http://localhost:8080/v2/send'   -d '{"message": "Test via Signal API!", "number": "+336XXXXXXXX", "recipients": [ "+336XXXXXXXX" ]}'

i have this error :
"error":"Error loading state file for user +336XXXXXXXX: Incorrect file format: expected parameter username not found  (InvalidObjectException)\n"}

i have the same result with this command line:
udo docker run -d --name signal-api --restart=always -p 8080:8080       -v $HOME/.local/share/signal-cli:/home/.local/share/signal-cli       -e 'MODE=normal' bbernhard/signal-cli-rest-api
can you help me?

**Comment by WallyCZ** [comment]
By me it was wrong platform (armv7 x arm64). Be sure that you are using arm64 (f.e. Librelec's docker downloads armv7 so you have to enforce to download arm64)

**Reply by mevrard31** [reply]
yes, i use a raspberry V4

# more /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
# lscpu
Architecture:                       aarch64
Byte Order:                         Little Endian
CPU(s):                             4
On-line CPU(s) list:                0-3
Thread(s) per core:                 1
Core(s) per socket:                 4
Socket(s):                          1
Vendor ID:                          ARM
Model:                              3
Model name:                         Cortex-A72
Stepping:                           r0p3
CPU max MHz:                        1500.0000
CPU min MHz:                        600.0000
BogoMIPS:                           108.00
L1d cache:                          128 KiB
L1i cache:                          192 KiB
L2 cache:                           1 MiB
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Vulnerable
Vulnerability Spectre v1:           Mitigation; __user pointer sanitization
Vulnerability Spectre v2:           Vulnerable
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fp asimd evtstrm crc32 cpuid
# more /proc/cpuinfo
processor       : 0
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 1
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 2
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 3
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

Hardware        : BCM2835
Revision        : c03112
Serial          : 10000000713c9242
Model           : Raspberry Pi 4 Model B Rev 1.2


i pull this container;
 docker pull bbernhard/signal-cli-rest-api@sha256:e4f9b0f0d0e70bd8b20167b09ded7b8ba6ecdf5a7f406f7eeb7467e387f92663

**Reply by mevrard31** [reply]
e4f9b0f0d0e7
linux/arm/v7
287.06 MB

**Comment by WallyCZ** [comment]
Try to force to use arm64 using `export DOCKER_DEFAULT_PLATFORM=linux/arm64` or use direct image id like this one: bbernhard/signal-cli-rest-api:latest@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03

**Comment by mevrard31** [comment]
don't work...

# export DOCKER_DEFAULT_PLATFORM=linux/arm64
# docker pull  bbernhard/signal-cli-rest-api:latest@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03                                                 docker.io/bbernhard/signal-cli-rest-api@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03: Pulling from bbernhard/signal-cli-rest-api
2378679266ac: Pull complete
4f855dc454f4: Pull complete
06cfbead4ab6: Pull complete
62ab5928b774: Pull complete
e0accfdafa18: Pull complete
3946bf6cc3bf: Pull complete
199d9f88add0: Pull complete
287304bfb57d: Pull complete
a7b537903145: Pull complete
c0c2b5f7c740: Pull complete
e239bae7ee15: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03
Status: Downloaded newer image for bbernhard/signal-cli-rest-api@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03
docker.io/bbernhard/signal-cli-rest-api:latest@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03
# export DOCKER_DEFAULT_PLATFORM=linux/arm64
# docker run -d --name signal-api --restart=always -p 8080:8080       -v $HOME/.local/share/signal-cli:/home/.local/share/signal-cli       -e 'MODE=native' bbernhard/signal-cli-rest-api:latest@sha256:659f7f356d87a893c5d3d1260a5c70847c81f58256c896e23d05d37e215e5e03
9b531060e4c4b79ba2d4683e51e56158a3bd6752153f2bd65e3029f7dd189317
# docker ps -a
CONTAINER ID   IMAGE                                  COMMAND            CREATED          STATUS                                     PORTS                                       NAMES
9b531060e4c4   bbernhard/signal-cli-rest-api:latest   "/entrypoint.sh"   12 seconds ago   Up Less than a second (health: starting)   0.0.0.0:8080->8080/tcp, :::8080->8080/tcp   signal-api
# docker ps -a
CONTAINER ID   IMAGE                                  COMMAND            CREATED          STATUS                           PORTS                                       NAMES
9b531060e4c4   bbernhard/signal-cli-rest-api:latest   "/entrypoint.sh"   17 seconds ago   Up 1 second (health: starting)   0.0.0.0:8080->8080/tcp, :::8080->8080/tcp   signal-api
# docker ps -a
CONTAINER ID   IMAGE                                  COMMAND            CREATED          STATUS                          PORTS     NAMES
9b531060e4c4   bbernhard/signal-cli-rest-api:latest   "/entrypoint.sh"   23 seconds ago   Restarting (159) 1 second ago             signal-api
# docker logs signal-api
null
